tests/test_contrib/test_hypothesis/test_laws/test_custom_strategy_for_callable.py

```diff
@@ -0,0 +1,161 @@
+        override the strategy easily. Overriding the strategy directly for
+        `KindN` is not currently possible.
+        """
+        container = make_container()
+        assert_equal(
+            some_identity.do_nothing(some_identity.do_nothing(container)),
+            some_identity.do_nothing(container),
+        )
+
+
+class _SomeIdentityN(
+    Lawful['_SomeIdentityN[_FirstType, _SecondType, _ThirdType]'],
+    Generic[_FirstType, _SecondType, _ThirdType],
+):
+    """Dummy interface that does nothing to an `Applicative` container."""
+
+    __slots__ = ()
+
+    _laws: ClassVar[Sequence[Law]] = (Law2(_LawSpec.idempotence_law),)
+
+    @abstractmethod  # noqa: WPS125
+    def do_nothing(
+        self,
+        container: Kind1[_F_Applicative, _ValueType],
+    ) -> Kind1[_F_Applicative, _ValueType]:
+        """No-op method that returns the container."""
+
+
+_SomeIdentity1: TypeAlias = _SomeIdentityN[_FirstType, Never, Never]
+
+
+class _Wrapper(
+    BaseContainer,
+    SupportsKind1['_Wrapper', _FirstType],
+    _SomeIdentity1[_FirstType],
+):
+    """Simple instance of `_SomeIdentityN`."""
+
+    _inner_value: _FirstType
+
+    def __init__(self, inner_value: _FirstType) -> None:
+        super().__init__(inner_value)
+
+    def do_nothing(
+        self,
+        container: Kind1[_F_Applicative, _ValueType],
+    ) -> Kind1[_F_Applicative, _ValueType]:
+        """No-op method that returns the container."""
+        return container
+
+
+def _callable_strategy(
+    arg1: type[object], arg2: type[object]
+) -> StrategyFactory[Callable]:
+    type_arg1 = int if arg1 == Any else arg1  # type: ignore[comparison-overlap]
+    type_arg2 = int if arg2 == Any else arg2  # type: ignore[comparison-overlap]
+    return_results = st.functions(
+        pure=True,
+        returns=strategy_from_container(Result)(Result[type_arg1, type_arg2]),  # type: ignore[valid-type]
+    )
+    return_maybes = st.functions(
+        pure=True,
+        returns=strategy_from_container(Maybe)(
+            MaybeLike2[type_arg1, type_arg2]  # type: ignore[valid-type]
+        ),
+    )
+    return st.one_of(return_results, return_maybes)
+
+
+def _callable_factory(thing: type[object]) -> StrategyFactory[Callable]:
+    if get_origin(thing) != Callable:
+        raise NotImplementedError
+
+    match get_args(thing):
+        case [[], return_type] if (
+            get_origin(return_type) == KindN
+            and get_args(return_type)[0] == _F_Applicative
+        ):
+            type1, type2, *_ = get_args(return_type)[1:]
+            return _callable_strategy(type1, type2)
+        case _:
+            raise NotImplementedError
+
+
+other_strategies: dict[type[object], StrategyFactory] = {
+    Callable: _callable_factory  # type: ignore[dict-item]
+}
+
+check_all_laws(
+    _Wrapper,
+    container_strategy=st.builds(_Wrapper, st.integers()),
+    other_strategies=other_strategies,
```

Comment:
maybe `register_strategies`? `other` feels a bit off to me.

Reply:
Agreed. How about `type_strategies`?